crates/rpc/eth/logs.rs

```diff
@@ -0,0 +1,183 @@
+        match params.as_deref() {
+            Some([param]) => {
+                let param = param.as_object().ok_or(RpcErr::BadParams)?;
+                let from_block = param
+                    .get("fromBlock")
+                    .ok_or_else(|| RpcErr::MissingParam("fromBlock".to_string()))
+                    .and_then(|block_number| BlockIdentifier::parse(block_number.clone(), 0))?;
+                let to_block = param
+                    .get("toBlock")
+                    .ok_or_else(|| RpcErr::MissingParam("toBlock".to_string()))
+                    .and_then(|block_number| BlockIdentifier::parse(block_number.clone(), 0))?;
+                let address_filters = param
+                    .get("address")
+                    .ok_or_else(|| RpcErr::MissingParam("address".to_string()))
+                    .and_then(|address| {
+                        match serde_json::from_value::<Option<AddressFilter>>(address.clone()) {
+                            Ok(filters) => Ok(filters),
+                            _ => Err(RpcErr::WrongParam("address".to_string())),
+                        }
+                    })?;
+                let topics_filters = param
+                    .get("topics")
+                    .ok_or_else(|| RpcErr::MissingParam("topics".to_string()))
+                    .and_then(|topics| {
+                        match serde_json::from_value::<Option<Vec<TopicFilter>>>(topics.clone()) {
+                            Ok(filters) => Ok(filters),
+                            _ => Err(RpcErr::WrongParam("topics".to_string())),
+                        }
+                    })?;
+                Ok(LogsRequest {
+                    from_block,
+                    address_filters,
+                    topics: topics_filters.unwrap_or_else(Vec::new),
+                    to_block,
+                })
+            }
+            _ => Err(RpcErr::BadParams),
+        }
+    }
+    // TODO: This is longer than it has the right to be, maybe we should refactor it.
+    // The main problem here is the layers of indirection needed
+    // to fetch tx and block data for a log rpc response, some ideas here are:
+    // - The ideal one is to have a key-value store BlockNumber -> Log, where the log also stores
+    //   the block hash, transaction hash, transaction number and its own index.
+    // - Another on is the receipt stores the block hash, transaction hash and block number,
+    //   then we simply could retrieve each log from the receipt and add the info
+    //   needed for the RPCLog struct.
+    fn handle(&self, storage: Store) -> Result<Value, RpcErr> {
+        let Ok(Some(from)) = self.from_block.resolve_block_number(&storage) else {
+            return Err(RpcErr::WrongParam("fromBlock".to_string()));
+        };
+        let Ok(Some(to)) = self.to_block.resolve_block_number(&storage) else {
+            return Err(RpcErr::WrongParam("toBlock".to_string()));
+        };
```

Comment:
```suggestion
        let Some(from) = self.from_block.resolve_block_number(&storage)? else {
            return Err(RpcErr::WrongParam("fromBlock".to_string()));
        };
        let Some(to) = self.to_block.resolve_block_number(&storage)? else {
            return Err(RpcErr::WrongParam("toBlock".to_string()));
        };
```
A `StorageError` here means failure to read from the DB, it doesn't mean that the parameter in the request is wrong. `StorageError`s get mapped to `RpcErr::Internal` which is the correct error to return for these cases.

Reply:
But what if we were given a block number higher than what exists? Shouldn't that be a wrong parameter? 
We can do the following:
1. Return WrongParam if we were given a block number.
2. Return an Internal error if we were given a tag.

Reply:
I think she is suggesting something like this 

```

        let block_number = self
            .from_block
            .resolve_block_number(&storage)?
            .ok_or(RpcErr::WrongParam("fromBlock".to_string())?;
```

If `resolve_block_number` returns an error, it will be sent as StoreError and converted into InternalError

Reply:
Changed [here](https://github.com/lambdaclass/ethereum_rust/pull/360/commits/218cbde30aa51dc4d569693e56e7d6fea181e775)